sigopt/xgboost/experiment.py

```diff
@@ -3,30 +3,41 @@
 import copy
 
 from .. import create_experiment
+from .defaults import (
+  SEARCH_BOUNDS,
+  SEARCH_PARAMS,
+  DEFAULT_SEARCH_PARAMS,
+  DEFAULT_BO_ITERATIONS,
+  DEFAULT_NUM_BOOST_ROUND,
+  DEFAULT_REGRESSION_METRICS,
+  DEFAULT_CLASSIFICATION_METRICS,
+  SUPPORTED_METRICS_TO_OPTIMIZE
+)
 
-DEFAULT_CLASSIFICATION_METRICS = ['accuracy', 'precision', 'recall', 'F1']
-DEFAULT_REGRESSION_METRICS = ['mean absolute error', 'mean squared error']
-DEFAULT_NUM_BOOST_ROUND = 10
 
 class XGBExperiment:
   def __init__(self, experiment_config, dtrain, evals, params, num_boost_round, run_options):
     self.experiment_config = experiment_config
+    self.experiment_config_parsed = None
     self.dtrain = dtrain
     self.evals = evals
     self.params = params
     self.num_boost_round = num_boost_round
     self.run_options = run_options
     self.sigopt_experiment = None
 
-  def parse_and_create_parameter_space(self):
-    # TODO: deal with later
-    pass
-
-  def parse_and_create_experiment(self):
-    experiment_config_parsed = copy.deepcopy(self.experiment_config)
+  def parse_and_create_metrics(self):
+    if 'metrics' in self.experiment_config_parsed and isinstance(self.experiment_config_parsed['metrics'], str):
+      assert self.experiment_config_parsed['metrics'] in SUPPORTED_METRICS_TO_OPTIMIZE
+      metric_to_optimize = self.experiment_config_parsed['metrics']
+      self.experiment_config_parsed['metrics'] = [{
+        'name': metric_to_optimize,
+        'strategy': 'optimize',
+        'objective': 'maximize'
```

Comment:
not always maximizing here, we want to min the regression metrics.

Reply:
So this is a slightly different behavior than how the auto-completion for parameter space works, since now the shortened version of metrics is a single string. Whereas for parameters, it's still a list of dicts. I think this might cause some confusion for the user.

Reply:
Still need to change this?